internals/daemon/api_checks.go

```diff
@@ -66,3 +70,49 @@ func v1GetChecks(c *Command, r *http.Request, _ *UserState) Response {
 	}
 	return SyncResponse(infos)
 }
+
+func v1PostChecks(c *Command, r *http.Request, _ *UserState) Response {
+	var payload struct {
+		Action string   `json:"action"`
+		Checks []string `json:"checks"`
+	}
+
+	decoder := json.NewDecoder(r.Body)
+	if err := decoder.Decode(&payload); err != nil {
+		return BadRequest("cannot decode data from request body: %v", err)
+	}
+
+	if len(payload.Checks) == 0 {
+		return BadRequest("must specify checks for %s action", payload.Action)
+	}
+
+	var err error
+	var checks []string
+	checkmgr := c.d.overlord.CheckManager()
+	plan := c.d.overlord.PlanManager().Plan()
+
+	switch payload.Action {
+	case "start":
+		checks, err = checkmgr.StartChecks(plan, payload.Checks)
+	case "stop":
+		checks, err = checkmgr.StopChecks(plan, payload.Checks)
+	default:
+		return BadRequest("invalid action %q", payload.Action)
+	}
+	if err != nil {
+		return BadRequest("cannot %s checks: %v", payload.Action, err)
+	}
+
+	st := c.d.overlord.State()
+	st.EnsureBefore(0) // start and stop tasks right away
+
+	var result string
+	if len(checks) == 0 {
+		result = fmt.Sprintf("No checks needed to %s", payload.Action)
```

Comment:
I know this is somewhat parallel to what services does with the change summary, but there we're limited by the structure of changes. Here, however, it seems like putting display logic or formatting in the API. I think it'd be better to return the slice of check names started/stopped directly, and let the client do what it wants to with them.

However, rather than return a `[]string` directly, let's be a bit more future-proof and return a `*ChecksActionResult` with a `Checks []string` field.

On the client side I'd probably simplify (drop mention of "queued") and just say something like:

- "Checks already started." (or stopped)
- "Checks started: foo, bar, baz."